gcovr/coverage.py

```diff
@@ -400,6 +411,9 @@ def decision_coverage(self) -> Tuple[int, int, int, Optional[float]]:
         return total, cover, unchecked, percent
 
 
+CovData = Dict[str, FileCoverage]
```

Comment:
The `covdata` has always been a dict, and I didn't intend to change that as part of this PR. But it would definitely be possible to introduce a `ProjectCoverage` class or something that.

What do you mean with debug output?

Reply:
Output like `Adding coverage for file...`

Also there where two places which add a new file coverage object if file isn‘t in dict. This can then be done while accessing the file the first time.

Reply:
I see your point.

Ok, I'll think about this and experiment a bit. But if I can't find a good design I'd rather keep the status quo and maybe revisit that point in the future. I'm trying to balance “whee fun refactoring” with “one step at a time, shipping something *good* is better than waiting until everything is *perfect*”.

One concern I have with adding such logging is that there is no clear “first” time. For example, consider the case of multiple worker threads. Initially, these load coverage data into separate CovData dicts. Thus, multiple workers might see the same file for the “first” time, if the same file is part of multiple compilation units. Doing this correctly would require something like a shared set of known filenames protected by a mutex, or a channel in which workers can publish file-processing events that are consumed by a separate thread. But that would be behaviour that is out of scope of the coverage data model.

Reply:
We can also finish this PR and do this later. After merging this and #597 we can fix the two bugs before creating the release.